### PR TITLE
LPS-107191 | Historical views charts

### PR DESCRIPTION
**Pull request by andresfulla**
@darquesdev

### CODE REVIEW

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/content/Language.properties [hunk 1]
```diff
@@ -13,4 +13,4 @@ this-number-refers-to-the-total-number-of-reads-since-the-content-was-published=
 this-number-refers-to-the-total-number-of-views-since-the-content-was-published=This number refers to the total number of views since the content was published.
 total-reads=Total Reads
 total-views=Total Views
-views-metric=Views
+views-metric=Views
```
Comment (dgarciasarai) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/content/Language.properties]: We should change this:
- views-metric=Views Metric
- views=Views (from portal-impl)

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/display/context/AnalyticsReportsDisplayContext.java [hunk 2]
```diff
@@ -113,6 +114,13 @@ public AnalyticsReportsDisplayContext(
 				"namespace",
 				_portal.getPortletNamespace(
 					AnalyticsReportsPortletKeys.ANALYTICS_REPORTS)
+			).put(
+				"languageTag",
```
Comment (dgarciasarai) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/display/context/AnalyticsReportsDisplayContext.java]: Sort alphabetically (before 'namespace')

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.es.js [hunk 3]
```diff
@@ -31,8 +31,7 @@ const CHART_SIZE = {height: 180, width: 280};
 function keyToTranslatedLabelValue(key) {
 	if (key === 'analyticsReportsHistoricalViews') {
 		return Liferay.Language.get('views-metric');
-	}
-	else {
+	} else {
```
Comment (dgarciasarai) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.es.js]: New line ;) (run SF)

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.es.js [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+		return value / 1000 + 'K';
+	}
+	return value;
+}
+
+/*
+ * It generates a set of functions used to produce
+ * internationalized date related content.
+ */
+const generateDateFormatters = key => {
```
Comment (dgarciasarai) [inline modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.es.js]: Extract functions to util? 🤔What do you think?

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/content/Language.properties (hunk 1 above)
Comment (andresfulla) [reply modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/content/Language.properties]: the problem is that views may refer to `Views` as `Vistas` not `Impresiones`. Views is specific term in web analytics. I'll talk to Julia

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/display/context/AnalyticsReportsDisplayContext.java (hunk 2 above)
Comment (andresfulla) [reply modules/dxp/apps/analytics/analytics-reports-web/src/main/java/com/liferay/analytics/reports/web/internal/display/context/AnalyticsReportsDisplayContext.java]: That was there!!


I'll do it.

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.es.js (hunk 4 above)
Comment (andresfulla) [reply modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/META-INF/resources/js/components/Chart.es.js]: They're only used at this component. So I don't see the need.

**Comment by CarlosBrichete** [comment]
Just started reviewing :)

File: modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/content/Language.properties (hunk 1 above)
Comment (andresfulla) [reply modules/dxp/apps/analytics/analytics-reports-web/src/main/resources/content/Language.properties]: Just talked to Julia. It makes sense to keep it contextualised here.

**Comment by andresfulla** [comment]
@wincent @jbalsas @izaera 
would any of you mind reviewing this commit https://github.com/dgarciasarai/liferay-portal/pull/161/commits/2affa85f78510f48e6cdae1280c28eab3a917215 ??

Thanks!!!

File: modules/dxp/apps/analytics/analytics-reports-web/build.gradle [hunk 5]
```diff
@@ -1,6 +1,7 @@
 buildCSS {
 	imports = [
-		new File(npmInstall.nodeModulesDir, "@clayui/css/src/scss")
+		new File(npmInstall.nodeModulesDir, "@clayui/css/src/scss"),
+		new File(npmInstall.nodeModulesDir, "clay-charts-shared/src/scss")
```
Comment (bryceosterhaus) [inline modules/dxp/apps/analytics/analytics-reports-web/build.gradle]: How come you add the css from `clay-charts-shared`? This should be specifically for the old billboard.js version of charts.

**Comment by dgarciasarai** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by andresfulla** [comment]
@bryceosterhaus we added that module because it was using a specific version of `lodash` (the portal global version of lodash is not enough for recharts). Also events had to be polyfilled for it.

**Comment by dgarciasarai** [comment]
ci:test:sf

**Comment by andresfulla** [comment]
ci:test:sf

**Comment by dgarciasarai** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by wincent** [comment]
We almost certainly don't want the `modules/apps/frontend-js/frontend-js-recharts/src/main/resources/META-INF/resources/.gitkeep` in there. There are zero such files in liferay-portal now outside of `modules/sdk`.

**Comment by andresfulla** [comment]
Just sent it here https://github.com/brianchandotcom/liferay-portal/pull/84450

**Pull request closed without merging**